File: mcp-server/src/tools/getCashflow.ts
import { z } from 'zod';
import { apiGet } from '../utils/apiClient.js';

export const getCashflowTool = {
  name: 'get_cashflow' as const,
  description: 'Get total income, total expenses, and net cashflow for a user. Optionally filter by date range using start_date and end_date in YYYY-MM-DD format.',
  parameters: z.object({
    user_id: z.number().default(1).describe('The user ID'),
    start_date: z.string().optional().describe('Start date (YYYY-MM-DD)'),
    end_date: z.string().optional().describe('End date (YYYY-MM-DD)'),
  }),
  execute: async (args: { user_id: number; start_date?: string; end_date?: string }) => {
    const params: Record<string, string> = { user_id: String(args.user_id) };
    if (args.start_date) params.start_date = args.start_date;
    if (args.end_date) params.end_date = args.end_date;

    const data = await apiGet('/api/cashflow', params);
    return JSON.stringify(data);
  },
};

File: mcp-server/src/tools/get_cashflow.py
import json

from fastmcp import FastMCP

from ..utils.api_client import api_get
from ..utils.auth import get_current_token

def register(mcp: FastMCP) -> None:
    @mcp.tool(
        description="Get total income, total expenses, and net cashflow for a user. "
        "Optionally filter by date range using start_date and end_date in YYYY-MM-DD format.",
    )
    async def get_cashflow(
        user_id: int = 1,
        start_date: str | None = None,
        end_date: str | None = None,
    ) -> str:
        token = get_current_token()
        params: dict[str, str] = {"user_id": str(user_id)}
        if start_date:
            params["start_date"] = start_date
        if end_date:
            params["end_date"] = end_date

        data = await api_get("/api/cashflow", params, token)
        return json.dumps(data)
